src/devices/video/upd7220.cpp

```diff
@@ -1003,13 +1008,25 @@ void upd7220_device::draw_char(int x, int y)
 {
 	int isize,psize;
 	uint16_t tile_data;
+	const char *system = this->mconfig().root_device().name();
 
 	LOG("uPD7220 char check: %d %d %02x %08x %d %d %02x\n",x,y,m_figs.m_dir,m_ead,m_figs.m_d,m_figs.m_dc,m_figs.m_figure_type);
 
-	/* QX10 may require upper bits for psize, VT240 requires the opposite */
 	isize = m_figs.m_d;
 	psize = m_figs.m_dc + 1;
 
+	/*
+	 * The QX10 set the direction to 0 and the perpendicular size in terms of
+	 * memory words when it is clearing lines on the screen. This seems to be
+	 * in direct contradiction to the datasheet which says the it should be set
+	 * in terms of pixels. It does however behave correctly on real HW, so this
+	 * works around this by modifing the psize value when it was initially set
+	 * to 40 words to be 640 pixels.
+	 */
+	if (strcmp(system, "QX-10") == 0 && m_figs.m_dir == 0 && psize == 40) {
```

Comment:
Please don't use strcmp, it's an ugly magic number and prone to breakage. Additionally I don't think this works for "QC-10" anyway.
 
So in theory the quick solution is to make a setter that defaults to `false` and gets explicitly set to `true` in _qx10_/_qc10_ machine config(s), in practice I believe there must be a more logical solution there depending on what it tries to setup, like a "zero" that should be treated as max size like already noted by the pc98 Madou Monogatari games

Reply:
Ok I'll fix this to use a setter instead. 

Yeah I'm sure there most be something about the chip that isn't documented or properly documented. I've spent some time looking at the assembly listing for the cp/m bios call that handles clearing the screen as well as staring it in the debugger and haven't been able to see anything obvious during the setup that would make it behave differently though.

Reply:
Updated the screen clearing patch so it shoudl behave according to the specifics of the datasheet without the need for special casing for the qx10.